engine/crates/operation-checks/src/schema.rs

```diff
@@ -71,23 +71,25 @@ pub struct SchemaField {
     pub(crate) field_name: String,
     /// The type of the field without any wrapping type (! and []).
     pub(crate) base_type: String,
-    pub(crate) wrappers: WrapperTypes,
+    pub(crate) wrappers: WrappingTypes,
 }
 
 impl SchemaField {
     pub(crate) fn render_type(&self) -> String {
         let mut result = self.base_type.clone();
-        let wrappers: Vec<_> = self.wrappers.iter_wrappers().collect();
 
-        for wrapper in wrappers.iter().rev() {
-            result = match wrapper {
-                WrapperType::List => format!("[{result}]"),
-                WrapperType::Required => format!("{result}!"),
-                WrapperType::RequiredList => format!("[{result}]!"),
+        for list in self.wrappers.iter_list_types() {
+            result = match list {
+                ListType::List => format!("[{result}]"),
+                ListType::NonNullList => format!("[{result}]!"),
             }
         }
 
-        result
+        if self.wrappers.inner_is_required() {
+            format!("{result}!")
+        } else {
+            result
+        }
```

Comment:
Should this not be done before the list wrappers if it's for the inner type?

Reply:
yes :see_no_evil: Weird that tests didn't catch it, I'll write that does.

Reply:
done and pushed